### PR TITLE
Use terraform-plugin-framework-timetypes for RFC3339 string attributes

### PR DESCRIPTION
**Pull request by bflad**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Reference: https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework-timetypes/timetypes

Will submit a feature request upstream for `time.Time` to/from implementations to remove more of the boilerplate `string` <-> `time.Time` conversions and error handling.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,6 +4,7 @@ go 1.19
 
 require (
 	github.com/hashicorp/terraform-plugin-framework v1.3.4
+	github.com/hashicorp/terraform-plugin-framework-timetypes v0.1.0
```
Comment (austinvalle) [inline go.mod:7]: You might be already thinking of this, but do we want to release a `v0.2.0` and include those upstream changes?

**Review by austinvalle — APPROVED** [review]
PR looks great, one note, should we add a changelog entry for this PR? Switching to this type internally introduces semantic equality logic that wasn't previously here

File: go.mod (hunk 1 above)
Comment (bflad) [reply go.mod:7]: Just released timetypes@v0.2.0 and will update this migration to remove (now extra) code.

**Comment by bflad** [comment]
> should we add a changelog entry for this PR? Switching to this type internally introduces semantic equality logic that wasn't previously here

Since there's no backing API for these resources that could potentially change the formatting of the value, I'm not sure if there's a way for the semantic equality logic to have any effect in that sense.

**Review by austinvalle — APPROVED** [review]

**Review by SBGoods — APPROVED** [review]

**Pull request merged**